backend/main/chapters/c07_nested_loops.py

```diff
@@ -399,14 +399,62 @@ def solution(self, players: List[str]):
             """
              )]
 
-    class crack_password_exercise(ExerciseStep):
+    class player_vs_player_bonus(ExerciseStep):
         """
 Well done! If you'd like a bonus challenge, remove the repetition and only output each pair of players once, e.g:
 
     Alice vs Bob
     Alice vs Charlie
     Bob vs Charlie
 
+        """
+
+        def solution(self, players: List[str]):
+            for i in range(len(players)):
+                for j in range(i + 1, len(players)):
+                    print(players[i], 'vs', players[j])
```

Comment:
This solution is shown to users once they get stuck, so I generally prefer it to be a more basic solution like the first one you show.

Another possible solution is to just change `!=` to `<` in the original solution:

```python
            for player1 in players:
                for player2 in players:
                    if player1 < player2:
                        print(player1, 'vs', player2)
```

Reply:
OK, I'll replace it with the basic one.